icu4c/source/test/intltest/numbertest_api.cpp

```diff
@@ -1926,6 +1928,197 @@ void NumberFormatterApiTest::unitCurrency() {
             u"123,12 CN¥");
 }
 
+void NumberFormatterApiTest::runUnitInflectionsTestCases(UnlocalizedNumberFormatter unf,
+                                                         const UChar *skeleton,
+                                                         const UChar *conciseSkeleton,
+                                                         const UnitInflectionTestCase *cases,
```

Comment:
I prefer if it takes only one test case at a time.

Reply:
In C++ array construction can be done nice and concisely - e.g. in my latest PR my tests are looking like this:

        const UnitInflectionTestCase meterCases[] = {
            // Check up on the basic values that the compound patterns below are
            // derived from:
            {"meter", "de", nullptr, 1, u"1 Meter"},
            {"meter", "de", "genitive", 1, u"1 Meters"},
            {"meter", "de", nullptr, 2, u"2 Meter"},
            {"meter", "de", "dative", 2, u"2 Metern"},
            {"mile", "de", nullptr, 1, u"1 Meile"},
            {"mile", "de", nullptr, 2, u"2 Meilen"},
            {"day", "de", nullptr, 1, u"1 Tag"},
            {"day", "de", "genitive", 1, u"1 Tages"},
            {"day", "de", nullptr, 2, u"2 Tage"},
            {"day", "de", "dative", 2, u"2 Tagen"},
            {"decade", "de", nullptr, 1, u"1\u00A0Jahrzehnt"},
            {"decade", "de", nullptr, 2, u"2\u00A0Jahrzehnte"},
        ...

In Java I believe there'd always be a need to place the "TestCase()" constructor around this, for example. In which case we might as well have `runTestCase("decade", "de", nullptr, 2, u"2\u00A0Jahrzehnte")` in Java: thus I'm agreeing it to be reasonable to diverge from C++ in this.